refactor: security 설정 추가, h2-console 활성화

zhtmr opened the pull request:
- application.yml : h2-console 활성화 설정
- SecurityConfig : h2-console url 접속 설정
- 게시글 엔티티 추가, 좋아요 엔티티 추가, 유저 엔티티에 연관관계 
<img width="1250" alt="image" src="https://github.com/TodayWhatTodo/backend/assets/48509269/c1ba2987-5d81-4205-8ef3-fb9bbc057f1a">

yudonggeun committed:
feat : 회원가입 서비스 구현, user session dto 객체 구현

yudonggeun committed:
feat : 회사 클래스 분리

yudonggeun committed:
Merge branch 'main' into company

yudonggeun committed:
feat : 프로필 이미지 경로 수정 기능 구현

yudonggeun committed:
feat : Follow entity 추가, 팔로우 기능 구현

zhtmr committed:
refactor: security 설정 추가, h2-console 활성화

zhtmr committed:
refactor: security 설정 추가, h2-console 활성화

zhtmr committed:
feat: 게시글 엔티티 추가, 좋아요 관리 엔티티 추가

zhtmr committed:
feat: 게시글 엔티티 추가, 좋아요 관리 엔티티 추가

zhtmr committed:
Refactor: 연관관계 편의 메서드 수정

zhtmr committed:
Refactor: 연관관계 편의 메서드 수정

yudonggeun commented on src/main/java/com/project/todayWhatToDo/post/domain/LikePost.java:
```diff
@@ -0,0 +1,37 @@
+
+    public void setPost(Post post) {
+        if(this.post != null) {
+            post.getLikePosts().remove(this);
```
getLikePost가 public 함수이기 다른 곳에서 변경할 가능성이 높아서 취약한 부분으로 보여요. 불변 리스트로 반환하는 방법을 도입하는 것은 어떨까요? 그리고 Post 엔티티에서는 영속성 관리를 하지 않아서 해당 코드가 작동하지 않을 것 같아요.

yudonggeun commented on src/main/java/com/project/todayWhatToDo/post/domain/Post.java:
```diff
@@ -0,0 +1,68 @@
+    @JoinColumn(name = "user_id")
+    private User user;
+
+    @OneToMany(mappedBy = "post")
```
LikePost 엔티티의 연관관계 필드가 FetchType.LAZY 가 아니기 때문에 연관된 모든 엔티티를 조회하는 쿼리가 생길 것 같아서 LAZY  옵션을 주는 것이 좋을 것 같습니다.

zhtmr replied:
LikePost 엔티티에 연관관계 필드 레이지 모드로 수정했습니다

yudonggeun commented on src/main/java/com/project/todayWhatToDo/post/domain/LikePost.java:
```diff
@@ -0,0 +1,37 @@
+    @JoinColumn(name = "post_id")
+    private Post post;
+
+    public void setPost(Post post) {
```
중간 테이블에서 추가하는 것보다 Post 나 user를 통해서 추가하는 방식을 사용하면 좋을 것 같아요.
post.addLike(user) or user.like(post)

yudonggeun commented on src/main/java/com/project/todayWhatToDo/user/domain/User.java:
```diff
@@ -52,6 +53,14 @@ public class User {
     private List<Career> careers = new ArrayList<>();
 
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<LikePost> likePosts = new ArrayList<>();
```
도메인 별로 분리했기 때문에 의존성을 최소화하는 것이 좋다고 생각해서 이 부분을 게시판 도메인에서 관리하는 것이 어떨까요?

zhtmr replied:
User - LikePost - Post 엔티티로 지금 관계를 맺어놓은건데  User 에서 저걸빼면 좋아요 누른 게시물을 못가져오지않나요?

yudonggeun replied:
oneToMany 관계는 jpa에서 객체로 접근하기 위한 관계 설정을 위한 기능이라서 테이블에서 외래키 관계는 중간 테이블인 likepost에서 manytoone을 통해서 설정이 되기 때문에 외래키 설정을 걱정하신 거라면 문제 없어 보여요. 

그리고 좋아요를 누른 게시물을 전체 조회하는 기능으로 구현을 하게 되면 문제가 없지만 좋아요를 누른 게시글이 많아진다면 한번에 모든 데이터를 가져오는 기능으로 개발하는 것 보다는 패이징 처리를 하는 것이 좋아보여요. 

그래서 jpa 연관관계를 통한 조회 기능 개발보다는 따로 쿼리를 통해서 조회하는 것이 좋을 것 같습니다.

 spring data jpa를 활용하면 페이징 처리를 아주 간단하게 구현할 수 있는데 그 기능을 사용하는 것이 어떨까요?

yudonggeun committed:
refactor : application.yml 주석 제거

yudonggeun committed:
refactor : application.yml 주석 제거

yudonggeun reviewed (commented):
리뷰입니다.

yudonggeun committed:
feat : 알림 삭제, 조회, 확인 기능 구현

yudonggeun committed:
refactor : UserServiceTest 테스트 구조 정리

yudonggeun committed:
feat : 팔로우 취소 기능 구현

yudonggeun committed:
feat : 팔로워/팔로잉 목록 조회 기능 구현, 팔로워/팔로잉 전체 카운트 구현

yudonggeun committed:
fix : follwer list 조회 메서드 이름 오타 수정

yudonggeun committed:
test : 팔로우 테스트 코드 추가

yudonggeun committed:
refactor : 사용하지 않는 import 제거

yudonggeun committed:
Merge branch 'follow' into dev

yudonggeun committed:
Merge pull request #14 from TodayWhatTodo/notify

feat : 알림 삭제, 조회, 확인 기능 구현

yudonggeun committed:
Merge branch 'main' into dev

yudonggeun committed:
Merge branch 'userservice-test-refactor' into profile-image-update

# Conflicts:
#	src/main/java/com/project/todayWhatToDo/user/domain/User.java
#	src/main/java/com/project/todayWhatToDo/user/service/UserService.java
#	src/test/java/com/project/todayWhatToDo/user/service/UserServiceTest.java

yudonggeun committed:
fix : merge 충돌 버그 수정

yudonggeun committed:
feat : 경력 수정 기능 구현

yudonggeun committed:
feat : 경력 삭제 구현

yudonggeun committed:
feat : 팔로우 CRUD 기능 구현

1. 팔로우 기능
2. 팔로우 취소 기능
3. 팔로워 목록 조회
4. 팔로잉 목록 조회
5. 팔로워 수 조회
6. 팔로잉 수 조회

yudonggeun committed:
Merge branch 'main' into user

# Conflicts:
#	src/main/java/com/project/todayWhatToDo/user/domain/Follow.java
#	src/main/java/com/project/todayWhatToDo/user/service/UserService.java
#	src/test/java/com/project/todayWhatToDo/user/domain/UserTest.java
#	src/test/java/com/project/todayWhatToDo/user/service/UserServiceTest.java

yudonggeun committed:
fix : 병합 버그 충돌 해결

yudonggeun committed:
refactor : UserService 책임 분리

yudonggeun committed:
fix : FollowServiceTest 버그 수정

yudonggeun committed:
feat : 프로필 조회 기능 구현

yudonggeun committed:
feat : 회원 탈퇴 기능 구현

yudonggeun committed:
feat : 알림 수신 설정 변경 기능 구현

yudonggeun committed:
feat : 유저 도메인 기능 구현

1. 프로필 이미지 경로 변경 구현
2. 경력 수정 기능 구현
3. 경력 삭제 구현
4. 프로필 조회 기능 구현
5. 회원 탈퇴 기능 구현
6. 알림 수신 설정 변경 기능 구현

zhtmr committed:
fix: user 엔티티 rebase main

zhtmr committed:
Merge remote-tracking branch 'origin/post' into post

zhtmr committed:
fix: user 엔티티 rebase main

zhtmr commented:
post 브랜치에서 작업중이였는데.. 
잘못 커밋된 파일을 리셋을 했어야 했는데 git rebase main 을 해버려서 main 브랜치에 작업된 커밋 내역이 전부 딸려와버렸네요....

<img width="803" alt="image" src="https://github.com/TodayWhatTodo/backend/assets/48509269/630d1f30-4de0-435a-be6f-c6d224df5dda">




user 엔티티에 잘못 작성했던 코드를 지우고 최신화하려고 했는데 저렇게 되버렸습니다. 

저렇게 되고 post 브랜치 삭제하고 다시 작업하려고 했더니 post 브랜치에서 작업했던 내역이 사라져버렸네요.....

zhtmr commented:
일단 cherry-pick 으로 다시 가져왔습니다.. 이 리퀘스트는 취소할게요

yudonggeun commented:
개발한 내용이 삭제된 이유는 commit 메시지를 보면 rebase를 하신 것으로 보이는데요. rebase는 merge랑 다르게 동작해서 그런 것 같습니다.
